mifos-passcode-cmp/src/commonMain/kotlin/auth/passcode/components/PasscodeLengthSwitch.kt

```diff
@@ -0,0 +1,181 @@
+                disabledContainerColor = Color.Transparent,
+            ),
+            enabled = passcodeLength == PasscodeLength.SIX_DIGIT,
+            contentPadding = PaddingValues(0.dp)
+        ) {
+            Text("4 digits")
+        }
+
+        AnimatedContent(
+            targetState = passcodeLength,
+            modifier = Modifier
+                .fillMaxHeight()
+                .fillMaxWidth()
+                .background(Color.Transparent)
+                .clip(RoundedCornerShape(40.dp)),
+            transitionSpec = {
+                val isGoingRight = targetState.length > initialState.length
+                slideInHorizontally(
+                    spring(
+                        stiffness = Spring.StiffnessLow
+                    ),
+                    initialOffsetX = {
+                        if(isGoingRight) it else -it
+                    }
+                ) togetherWith slideOutHorizontally(
+                    spring(
+                        stiffness = Spring.StiffnessLow
+                    ),
+                    targetOffsetX = {
+                        if(isGoingRight) -it else it
+                    }
+                )
+            },
+            content = { passcodeLength ->
+                if(passcodeLength.length==6){
+                    SlidingTab(
+                        label ="6 digits",
+                        labelColor = disabledTextColor,
+                        color = tabColor,
+                        alignment = Alignment.CenterEnd
+                    )
+                }else{
+                    SlidingTab(
+                        label ="4 digits",
+                        labelColor = disabledTextColor,
+                        color = tabColor,
+                        alignment = Alignment.CenterStart
+                    )
+                }
+            },
+
+            label = ""
+        )
+
+
+
+        Button(
+            {
+                passcodeLength = PasscodeLength.SIX_DIGIT
+                onSelectSixDigit()
+            },
+            modifier = Modifier
+                .fillMaxWidth(.5f)
+                .fillMaxHeight()
+                .clip(
+                    RoundedCornerShape(
+                        40.dp
+                    )
+                )
+                .align(Alignment.CenterEnd),
+            colors = ButtonDefaults.buttonColors(
+                containerColor = Color.Transparent,
+                contentColor = enabledTextColor,
+                disabledContentColor = Color.Transparent,
+                disabledContainerColor = Color.Transparent,
+            ),
+            contentPadding = PaddingValues(0.dp),
+            enabled = passcodeLength == PasscodeLength.FOUR_DIGIT,
+        ) {
+            Text("6 digits")
+        }
+
+    }
+}
+
+@Composable
+private fun SlidingTab(
+    label: String = "",
+    color: Color = Color.Black,
+    labelColor: Color = Color.White,
```

Comment:
No need to add defaults as we are passing them from above `composable` for both case.
```
@Composable
private fun SlidingTab(
    label: String,
    color: Color,
    labelColor: Color,
    alignment: Alignment
) {
```